src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MIN_CAPACITY = 10;
+    private static final double GROWS_COEFFICIENT = 1.5;
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        elements = (T[]) new Object[MIN_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (isCapacityReached()) {
+            grow();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index + " is out of bounds!");
```

Comment:
You can put exception message into constant to provide DRY rule

Reply:
I don't think DRY is relevant here, as I only have 1 exception in a different method. So there is no repetetive code here 🙃